plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/DecoratorValidationProviderCDI20.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package org.jboss.tools.cdi.reddeer.validators;
+
+import java.util.Arrays;
+
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.jboss.tools.cdi.reddeer.annotation.ValidationType;
+
+public class DecoratorValidationProviderCDI20 extends AbstractValidationProvider {
```

Comment:
Same thing like in another PRs. If the problem messages repeats in whole classes, it should be extracted and defined once. JSR string can be derived in child class and prototype defined in parent as abstract method.

Reply:
Fixed.